frontend/src/store/features/bank/bankSlice.ts

```diff
@@ -0,0 +1,206 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { SendMsg } from '../../../txns/bank';
+import bankService from './bankService';
+import { signAndBroadcast } from '../../../utils/signing';
+import { Coin } from 'cosmjs-types/cosmos/base/v1beta1/coin';
+import { KeyLimitPagination, Msg } from '../../../types/types';
+import { TxStatus } from '../../../types/store';
+
+interface Balance {
+  list: Coin[];
+  status: TxStatus;
+  errMsg: string;
+}
+interface BankState {
+  balances: Record<string, Balance>;
+  tx: {
+    status: TxStatus;
+  };
+  multiSendTx: {
+    status: TxStatus;
+  };
+}
+
+const initialState: BankState = {
+  balances: {},
+  tx: {
+    status: TxStatus.INIT,
+  },
+  multiSendTx: { status: TxStatus.INIT },
+};
+
+export const getBalances = createAsyncThunk(
+  'bank/balances',
+  async (data: {
+    baseURL: string;
+    address: string;
+    chainID: string;
+    pagination: KeyLimitPagination;
+  }) => {
+    const response = await bankService.balances(
+      data.baseURL,
+      data.address,
+      data.pagination
+    );
+    return {
+      chainID: data.chainID,
+      data: response.data,
+    };
+  }
+);
+
+export const multiTxns = createAsyncThunk(
+  'bank/multi-txs',
+  async (
+    data: {
+      baseURL: string;
```

Comment:
Instead of defining it like this, create an interface with parameters that you can use in most places.